### ISSUE TITLE
Creating config for ThinkBook 15 G2 ARE

### ISSUE THREAD

**Issue by Zireael61** [hirschmann/nbfc#1253]
Hi, I am trying to create a config for ThinkBook 15 G2 ARE with Ryzen 4700U. I couldn't be able to find my EC by looking motherboard. I tried probing but I couldn't be able to change fan speed. I think I should activate manual control mode. How can I find a way to activate it? Is there a more detailed guide about how to create a config?

**Comment by Zireael61** [comment]
I have two different probing log, there were many register but I find only five of them are responding to temp and fan speed. The upper one is increasing fan speed & temp, the lower one is decreasing fan speed & temp.

0x0B: 2A   ,29,2A         ,2B      ,2C,2D
0xB0: 2F,32,38,3B,3E,40,42,43,45   ,59,60
0xB1: 2F,32,38,3B,3E,40,42,43,45   ,59,60
0xB2: 2F,2E   ,2F,31,32   ,33   ,34,35,37
0xB3: 34   ,37,3B,3F,41,43,45,46,47,53,60


0x0B: 34,35      ,34,33,32,31,30,2F,2E,2D   ,2C
0xB0: 59,62,5C,53,4B,43,3B,3D,35,36,38,30,31,34,36
0xB1: 59,62,5C,53,4B,43,3B,3D,35,36,38,30,31,34,36
0xB2: 38      ,36,34,32,31,30   ,2F   ,2E,2D   ,2C
0xB3: 59,62,5C,53,4F,49,41,40,3A,39,3A,35,34,36,38


Also, I found 0x10 is changing when I change profile using Lenovo Vantage. 00 for performance mode, 04 for battery saver, 20 for intelligent cooling. Unfortunately, it doesn't affect anything when I changed this register manually (However it stays the same as what I write, not go to the original value automatically)

**Comment by Zireael61** [comment]
I did a test. If I write some value to 0xB2, it will be overwritten very fastly by my laptop.
I made a config and set EC poll interval as 100 ms. If I write some high value for example 80 to 0xB2 with the 100 ms setting, my fans are starting to turn faster but my CPU becomes veeeery slow. I tried some lower values but this situation did not change. If my fans start to turn faster, my CPU always gets slower.

If I write 79 to 0xB2, it has no effect. If I write something greater than 79, my fan turns faster. There are no difference between 0-79 or 80-255

Edit: I am mistaken. When I set 80 on 0xB2, my fan speeds up also my cpu does not slow down. However, it has no effect on temps, interesting.
